app/middlewares/auth.js

```diff
@@ -0,0 +1,29 @@
+const errors = require('../errors'),
+  logger = require('../logger'),
+  jwt = require('jwt-simple'),
+  config = require('../../config/index'),
+  User = require('../models').users;
+
+exports.checkCredentials = (req, res, next) => {
+
+  if (!req.headers.token){
+    return res.status(401).send('You must be logged in to access this endpoint');
+  }
+
+  const token = jwt.decode(req.headers.token, config.common.session.secret);
+
+  User.findOne({
+    where: {
+      email: token.token
+    }
+  }).then(result => {
+
+    if(!result){
+      return res.status(401).send('Invalid credentials');
+    }
+    req.isAdmin = result.isAdmin;
```

Comment:
In particular I would implement 2 different middlewares, one that checks that the user is authenticated.
And another one that assuming that the user is already authenticated we check if is an user or an admin. So you can chain them together.